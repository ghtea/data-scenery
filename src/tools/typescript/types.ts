

// I had wrote this, but I forgot what it is...
export type UnionSuggesting<T extends U, U = string> = T | (U & {});
