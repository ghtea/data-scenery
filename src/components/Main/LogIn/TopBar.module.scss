@import 'styles/every.scss';

.root {

    width: auto; 
    height: 50px;
    
    position: absolute;
    top: 5px;

  
}


.home {

    display: flex;
    flex-flow: row nowrap;
    justify-self: flex-start;
    align-items: center;
    
    width: auto;
    height: 100%;

    font-size: 1rem;
  
    & > *:nth-child(1) {
        width: auto;

        
        .icon-logo {
            @include color-each-theme(color, main__5);
            width: 36px;
            height: 36px;
        }
    }

    & > *:nth-child(2) {
        width: auto;
        @include color-each-theme(color, basic__8);
    
        font-size: 1.4rem;
        font-weight: bold;
        
        margin-left: 10px;
    }


}



