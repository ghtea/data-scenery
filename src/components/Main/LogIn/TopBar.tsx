import React, { useCallback, useEffect } from "react";

import history from 'historyApp';
import { useLocation } from "react-router-dom";
import { FormattedMessage } from 'react-intl';

//import {useSelector, useDispatch} from "react-redux";
import useLink from 'tools/hooks/useLink';

import IconLogo from 'svgs/others/IconLogo';

import styles from './TopBar.module.scss';


type PropsTopBar = {};

function TopBar({}: PropsTopBar) {
  
  // event: React.MouseEvent<HTMLAnchorElement> | React.MouseEvent<HTMLButtonElement>, 
   
  const {onClick_LinkInsideApp} = useLink(history);

  return (
    <div 
        className={`${styles['root']}`}
    >
        <a  
            className={`${styles['home']}`}
            href='/' 
            onClick={onClick_LinkInsideApp}
        >
            <div>
                <IconLogo className={`${styles['icon-logo']}`} />
            </div>
            
            <div> 
                <FormattedMessage id={`Nav.NameApp`} />
            </div>
        </a>
    </div>
  );
}

TopBar.defaultProps = {};

export default TopBar;
