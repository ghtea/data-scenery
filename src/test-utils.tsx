import React from 'react';
import { render as renderRtl, RenderOptions } from '@testing-library/react';

import { IntlProvider } from 'react-intl';
import translationEn from 'language/translation/en.json';

import { Router} from "react-router-dom";  // BrowserRouter
import history from 'historyApp';

import {Provider, useSelector, useDispatch} from 'react-redux' 
import store from './store';

import { CookiesProvider } from 'react-cookie';



// wrote automatically from IDE tooltip
type Ui = React.ReactElement<any, string | ((props: any) => React.ReactElement<any, string | any | (new (props: any) => React.Component<any, any, any>)> | null) | (new (props: any) => React.Component<any, any, any>)>;
type Option = Pick<RenderOptions<typeof import("@testing-library/dom/types/queries")>, "container" | "baseElement" | "hydrate" | "wrapper"> | undefined;


const wrapper:React.FunctionComponent<{}> = ({ children }) => {
    return (
        <Router history={history}>
        <CookiesProvider>
        <Provider store={store}>
        <IntlProvider locale={'en'} messages={translationEn} >
            {children}
        </IntlProvider>
        </Provider>
        </CookiesProvider>
        </Router>
    )
}

const render = (ui: Ui, option?: Option) =>
    renderRtl(ui, { wrapper: wrapper, ...option })



// re-export everything
export * from '@testing-library/react';

// override render method
export { render }
