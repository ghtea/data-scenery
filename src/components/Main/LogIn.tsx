import React, { useCallback, useEffect, useState } from "react";
import firebaseApp, { firebaseAuth } from 'firebaseApp';
import { useLocation } from "react-router-dom";
import history from 'historyApp';

import { FormattedMessage, useIntl } from 'react-intl';

import {useSelector, useDispatch} from "react-redux";
import {StateRoot} from 'store/reducers';

import * as actionsAuth from 'store/actions/auth';
import * as actionsStatus from 'store/actions/status';

import useInputBasic from 'tools/hooks/useInputBasic';

//import IconLogIn from 'svgs/basic/IconLogIn';
import TopBar from './LogIn/TopBar';

import styles from './LogIn.module.scss';




type PropsLogIn = {};

function LogIn({}:PropsLogIn) {
  
    const dispatch = useDispatch();
    const intl = useIntl();
  
    const readyUser = useSelector((state: StateRoot) => state['status']['ready']['user']);
    const loadingUser = useSelector((state: StateRoot) => state['status']['loading']['user']);

    // when login button is pushed, notification code of reaction is added to  this list, when login button is pushed again this list cleared once 
    const listCodeSituationOthers = useSelector((state: StateRoot) => state['notification']['listCodeSituationOthers']);

  
    const onClick_LinkInsideApp = useCallback(
        (event: React.MouseEvent<HTMLAnchorElement> | React.MouseEvent<HTMLButtonElement>, destination:string) => {
        history.push(destination);
        },[history]
    );
    
    const {draft: draft_Main, onChange: onChange_Main} = useInputBasic({
        emailAddress: '',
        password: '' 
    });


    const [codeSituationEmail, setCodeSituationEmail] = useState('');
    const [codeSituationPassword, setCodeSituationPassword] = useState('');
  
    useEffect(()=>{
        if (readyUser) {
            history.push('/');
        }
    },[readyUser, loadingUser]);

    useEffect(()=>{
        if(listCodeSituationOthers.includes('LogIn_NoEmail__E')){
            setCodeSituationEmail('LogIn_NoEmail__E');
            setCodeSituationPassword('');
        }
        else if(listCodeSituationOthers.includes('LogIn_InvalidEmail__E')){
            setCodeSituationEmail('LogIn_InvalidEmail__E');
            setCodeSituationPassword('');
        }
        else if (listCodeSituationOthers.includes('LogIn_NoPassword__E')) {
            setCodeSituationEmail('');
            setCodeSituationPassword('LogIn_NoPassword__E');
        }
        else if (listCodeSituationOthers.includes('LogIn_WrongPassword__E')) {
            setCodeSituationEmail('');
            setCodeSituationPassword('LogIn_WrongPassword__E');
        }
        else {
            setCodeSituationEmail('');
            setCodeSituationPassword('');
        }
    },[listCodeSituationOthers])


    const submitMain = useCallback(
        () => {
            dispatch(actionsAuth.return__LOG_IN({
                email: draft_Main.emailAddress,
                password: draft_Main.password
            }));
        },
        [draft_Main]
    );
    const onSubmit_Main = useCallback(
        (event:React.FormEvent<HTMLFormElement>) => {
            event.preventDefault();
            submitMain();
        },
        []
    );
    const onKeyPress_Main = useCallback(
        (event: React.KeyboardEvent<HTMLInputElement>) => {
            if (event.key === "Enter") {
                submitMain();
            }
        },
        []
    );
    

    const onClick_LogInSocial = useCallback(
        (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
            const {currentTarget: {value}} = event;
            if (value === 'google'){
                dispatch(actionsAuth.return__LOG_IN_GOOGLE() );
            }
            else if (value === 'github'){
                dispatch(actionsAuth.return__LOG_IN_GITHUB() );
            }
        }, []
    );
  
  return (
    <div className={`${styles['root']}`} >

        <TopBar />

        <form 
            className={`${styles['content']}`} 
            onSubmit={onSubmit_Main}
        >

            <div className={`${styles['title-page']}`} > 
                <FormattedMessage id={`Main.LogIn_LogIn`} />
            </div>
                
            <div className={`${styles['input-identity']}`} >
                <div> <FormattedMessage id={`Main.LogIn_EmailAddress`} /> </div>
                <input 
                    type='email'
                    placeholder={intl.formatMessage({ id: 'Main.LogIn_EmailAddress'})}
                    name='emailAddress'
                    value={draft_Main.emailAddress}
                    required
                    onChange={onChange_Main} 
                    onKeyPress={onKeyPress_Main}
                /> 
                <div> { codeSituationEmail && <FormattedMessage id={`Notification.${codeSituationEmail}`} /> }  </div>
            </div> 
                
            <div className={`${styles['input-password']}`} >
                <div> <FormattedMessage id={`Main.LogIn_Password`} /> </div>
                <input 
                    type='password'
                    placeholder={intl.formatMessage({ id: 'Main.LogIn_Password'})}
                    name='password'
                    value={draft_Main.password}
                    required
                    onChange={onChange_Main}
                    onKeyPress={onKeyPress_Main}
                /> 
                <div> { codeSituationPassword && <FormattedMessage id={`Notification.${codeSituationPassword}`}/>} </div>
            </div> 

            
            <div className={`${styles['button-enter']}`} >
                <input
                    type='submit'
                    value={intl.formatMessage({ id: 'Main.LogIn_LogIn'})}
                />
            </div> 
            
        
            <div className={`${styles['division']}`} > 
                <div> or Log In with </div>
            </div> 
            
            <div className={`${styles['collection-social']}`} >
                <button 
                    type='button'
                    value='google'
                    onClick={onClick_LogInSocial}
                > Google </button>
                <button 
                    type='button'
                    value='github'
                    onClick={onClick_LogInSocial}
                > GitHub </button>
            </div> 
            
            <nav className={`${styles['collection-link']}`} >
                <div> 
                    <a  
                        href='/'
                    > <FormattedMessage id={`Nav.Home`} /> </a> 
                </div>
                <div> 
                    <a  
                        href='/sign-up'
                    > <FormattedMessage id={`Nav.SignUp`} /> </a> 
                </div>
            </nav> 

        </form>
    </div>
  );
}

LogIn.defaultProps = {};

export default LogIn;

