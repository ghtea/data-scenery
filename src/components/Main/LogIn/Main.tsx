import React from 'react';
import { Route, Switch } from "react-router-dom";

import Home from "../Home";
import LogIn from "../LogIn";
import SignUp from "../SignUp";
import Test from "../Test";
import NotFound from "../NotFound";

//import SignUp from "./components/Auth/SignUp";

import styles from './Main.module.scss';


type PropsMain = {};

function Main({}: PropsMain) {
  return (
    
    <main className={`${styles['root']}`}>
          <Switch>
            
            <Route exact path="/" >
                <Home />
            </Route>

            <Route path="/log-in" >
                <LogIn />
            </Route>

            <Route path="/sign-up" >
                <SignUp />
            </Route>

            <Route path="/test" >
                <Test />
            </Route>

            <Route >
                <NotFound />
            </Route>

          </Switch>
        
      </main>
  );
}

export default Main;

