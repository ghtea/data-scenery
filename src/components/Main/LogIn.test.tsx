import React from 'react';
import { render, fireEvent } from 'test-utils';
// import '@testing-library/jest-dom/extend-expect';    // 이거 필요없는 듯?

import LogIn from './LogIn';
import { StringLiteral } from 'typescript';


describe('<LogIn />', () => {


    it('matches snapshot', () => {
        const utils = render(<LogIn />);
        expect(utils.container).toMatchSnapshot();
    });

    it('shows the props correctly', () => {
        const utils = render(<LogIn />);
        fireEvent.change(utils.getByRole('image', { name: 'fancy image' }))

        // utils.getByRole('image', { name: 'fancy image' })
    });

});
