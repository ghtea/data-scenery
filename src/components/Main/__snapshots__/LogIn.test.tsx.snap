// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> matches snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="root"
    >
      <div>
        <div
          class="icon-logo icon"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            focusable="false"
            height="100%"
            role="img"
            viewBox="0 0 640 512"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M635.73 406.91l-194.04-297.6C435.9 100.44 425.95 96 416 96c-9.95 0-19.9 4.44-25.69 13.31l-52 79.76-70.79-110.55C261.32 68.84 250.66 64 240 64s-21.32 4.84-27.52 14.52L4.58 403.18C-7.99 422.81 6.81 448 30.92 448h580.22c22.5 0 36.32-23.09 24.59-41.09zM63.61 400L240 124.55 416.39 400H63.61zm409.78 0L366.71 233.4 416 157.8 573.92 400H473.39z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div>
        <a>
          Data Scenery
        </a>
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="title-page"
      >
        Log In
      </div>
      <div
        class="input-identity"
      >
        <div>
           
          Email Address
           
        </div>
        <input
          placeholder="Email Address"
          required=""
          type="email"
          value=""
        />
        <div>
           
          
            
        </div>
      </div>
      <div
        class="input-password"
      >
        <div>
           
          Password
           
        </div>
        <input
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <div>
           
          
           
        </div>
      </div>
      <div
        class="button-enter"
      >
        <button>
           
          Log In
        </button>
      </div>
      <div
        class="division"
      >
        <div>
           or Log In with 
        </div>
      </div>
      <div
        class="collection-social"
      >
        <button
          name="google"
        >
           Google 
        </button>
        <button
          name="apple"
        >
           Apple 
        </button>
        <button
          name="github"
        >
           GitHub 
        </button>
      </div>
      <div
        class="collection-link"
      >
        <div>
          <a>
             
            Home
             
          </a>
        </div>
        <div>
           
          <a>
             
            Sign Up
             
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
